packages/widget/src/stores/form/useFieldActions.ts

```diff
@@ -1,8 +1,48 @@
+import { useCallback } from 'react';
 import { shallow } from 'zustand/shallow';
-import type { FormActions } from './types.js';
+import type { widgetEvents } from '../../hooks/useWidgetEvents.js';
+import { useWidgetEvents } from '../../hooks/useWidgetEvents.js';
+import { WidgetEvent } from '../../types/events.js';
+import type {
+  DefaultValues,
+  FormActions,
+  FormFieldNames,
+  GenericFormValue,
+  SetOptions,
+} from './types.js';
 import { useFormStore } from './useFormStore.js';
 
+interface FieldValueToEmittedEvents {
+  [key: string]: (
+    value: GenericFormValue,
+    emitter: typeof widgetEvents,
+  ) => void;
+}
+
+const fieldValueToEmittedEvents: FieldValueToEmittedEvents = {
+  toAddress: (address, emitter) =>
```

Comment:
Let's add all fields here to avoid adding them in the future anyway.

Reply:
Lets go with one event for them all. It's simpler that way

Reply:
Just gonna add in better type inference too